src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -2668,11 +2670,11 @@ void Lowering::TryLowerCselToCinc(GenTreeOp* select, GenTree* cond)
                     // This would remove gains from this optimisation so don't proceed.
                     return;
                 }
-                select->gtOp2    = select->gtOp1;
                 GenTree* revCond = comp->gtReverseCond(cond);
                 assert(cond == revCond); // Ensure `gtReverseCond` did not create a new node.
             }
-            select->gtOp1 = cond->AsOp();
+            BlockRange().Remove(select->gtOp2, true);
```

Comment:
Yeah, it's hard for the condition to interfere with the operands when we only do this optimization for constant operands. Interference between the condition and operands is what causes this case to occur.

If we generalized it to handle general `SELECT(cond, x, ADD(y, 1))` then I think it would be easier to hit. But I don't think we need to worry about that as part of this PR.